### PR TITLE
Copy Editing PR with a few content additions

### PR DESCRIPTION
**Pull request by R-Morgan**
Most of these commits are my copy editing suggestions for the Cabal section. I've tried to batch smaller, similar punctuation-based changes into single commits, while reworked snippets of the prose tend to have their own, smaller commits for easier acceptance/rejection.

There are a few commits that are a more additions to the Other Languages section that I wrote before we touched base about commit and PR structure.

Hope this is helpful!

### CODE REVIEW

File: tutorial.md [hunk 1]
```diff
@@ -12021,7 +12025,23 @@ because the terms have no universal meaning.
 Haskell
 -------
 
-Haskell's main implementation is *ghc*.
+Haskell's genesis happened in 1987 at the [Functional Programming Languages and
```
Comment (sdiehl) [inline tutorial.md]: This is nice.

**Comment by sdiehl** [comment]
This all looks good to me, is this complete enough that you'd like me to merge it?

**Comment by R-Morgan** [comment]
Yeah, I'd say so. I added one more commit to add some links. I got the most salient issues, and I'm not seeing anything else at the moment. Tomorrow, I'm planning on moving on to the Stack section.

**Comment by sdiehl** [comment]
Ok great, thanks for doing this.

**Pull request merged**